contribute/design/cstor-operator/specs.md

```diff
@@ -0,0 +1,423 @@
+	PoolSpec CStorPoolAttr              `json:"poolSpec"`
+}
+
+// CStorPoolClusterNodeSpec is the spec for node where pool should be created.
+type CStorPoolClusterNodeSpec struct {
+
+    // Name is the name of the node.
+	Name string `json:"name"`
+
+	// PoolSpec is the pool related specification that is used to provision cstor pool on the node.
+	PoolSpec CStorPoolClusterSpecAttr `json:"poolSpec"`
+
+	// BlockDeviceGroups contains the list of BD groups that should be used for pool provisioning on that node.
+	BlockDeviceGroups []CStorPoolClusterBlockDevieGroups `json:"groups"`
+}
+
+// CStorPoolClusterBlockDeviceGroups contains details of a block device group.
+type CStorPoolClusterBlockDeviceGroups struct {
+
+	// Name is the name of the block device group.
+	Name string `json:"name"`
+
+	// BlockDevices is the block devices present in the group.
+	BlockDevices []CStorPoolClusterBlockDevice `json:"blockDevice"`
+}
+
+// CStorPoolClusterBlockDevice contains the name details of a block device.
+type CStorPoolClusterBlockDevice struct {
+	// Name is the name of the block device.
+	Name string `json:"name"`
+
+	// ID is a unique id associated to this block device either by user or system.
+	// This ID does not change even if the block device is replaced so one can think it as a slot for this block device.
+	ID string `json:"id"`
+}
+
+// CStorPoolClusterStatus is for handling status of pool.
+type CStorPoolClusterStatus struct {
+	Phase string `json:"phase"`
+}
+
+// CStorPoolClusterSpecAttr is to describe zpool related attributes.
+type CStorPoolClusterSpecAttr struct {
+	CacheFile        string   `json:"cacheFile"`        //optional, faster if specified
+	PoolType         PoolType `json:"poolType"`         //mirrored, striped, raidz or raidz2
+	OverProvisioning bool     `json:"overProvisioning"` //true or false
+}
+
+
+```
+
+#### Spec for CStorPool
+TODO: Put sample YAMLs
+
+```
+
+
+// CStorPool describes a cstor pool resource created as custom resource.
+type CStorPool struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   CStorPoolSpec   `json:"spec"`
+	Status CStorPoolStatus `json:"status"`
+}
+
+// CStorPoolSpec is the spec listing fields for a CStorPool resource.
+type CStorPoolSpec struct {
+
+	// Group is the block device group.
+        Group    []BlockDeviceGroup   `json:"group"`
+
+        // PoolSpec is specification of pool.
+	PoolSpec CStorPoolAttr `json:"poolSpec"`
+}
+
+// BlockDeviceGroup contains a collection of block devices for a given pool topology in CSP.
+type BlockDeviceGroup struct {
+
+	// Item contains a list of CspBlockDevices.
+	Item []CspBlockDevice `json:"blockDevices"
+
+        // Name is the name of the block device group.
+	Name string    `json:"name"`
+}
+
+// CspBlockDevice contains the details of block device present on CSP.
+type CspBlockDevice struct {
+
+	// ID is the id where this block device is put in cspc
```

Comment:
ID is a unique identifier of this block device. A block device may get replaced while this ID remains same throughout the lifetime of this pool. One can consider this as a slot number for this block device.